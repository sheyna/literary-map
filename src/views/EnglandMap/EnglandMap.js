import { useState } from 'react';
import MapElement from '../../components/MapElement/MapElement';
import InfoBox from '../../components/InfoBox/InfoBox';
import locations from '../../locations.json';
import styles from './EnglandMap.module.css';

function EnglandMap() {
	const [ infoBoxOpen, setInfoBoxOpen ] = useState(false);
	const [ infoBoxLocation, setInfoBoxLocation ] = useState({});

	function openInfoBox (open, location) {
		setInfoBoxOpen(open);
		setInfoBoxLocation(location);
	}
	
	return (
		<>
			<div className={styles.mapElement}>
				<MapElement 
					locations={locations} 
					openInfoBox={openInfoBox}
				/>
			</div>
			<div className={styles.pageContents}>
				{infoBoxOpen
					&& <InfoBox 
						infoBoxLocation={infoBoxLocation} 
						key={infoBoxLocation.title} 
						openInfoBox={openInfoBox}
					/> 
				}
				<p className={styles.introText}>For such a small county, England has produced hundreds of the great works of literature. From Latin and Anglo-Norman literatures to the early development of the English language, stage plays and the birth of the novel, England has seen many types of great literature.</p>
			</div>
		</>
	);
	
}

export default EnglandMap;
