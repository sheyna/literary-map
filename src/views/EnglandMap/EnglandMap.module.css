.mapElement {
  left: 0;
  width: 100%;
  position: relative;
}

.pageContents {
  display: block;
  position: static;
  width: 80%;
  margin: 0 auto;
}

@media screen and (min-width: 640px) {
  .mapElement {
    width: 90%;
  }
}

@media screen and (min-width: 760px) {
  .pageContents {
    position: relative;
  }
  .mapElement {
    width: 100%;
  }
}

@media screen and (min-width: 1025px) {
  .mapElement {
    width: 70%;
    display: inline-block;
    max-width: 800px;
  }
  .pageContents {
    box-sizing: border-box;
    display: inline-block;
    width: 40%;
    margin: 30vw 0 0 0;
  }
  .introText {
    margin-right: 10%;
  }
}

@media screen and (min-width: 1450px) {
  .pageContents {
    margin-top: 35vw;
    width: 45%;
  }
}
