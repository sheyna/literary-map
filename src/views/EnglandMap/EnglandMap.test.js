import ReactTestRenderer from 'react-test-renderer';
import EnglandMap from  './EnglandMap';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

const testLocation = {
	"title": "Elizabeth Gaskell's Home",
	"book": "Mary Barton, North and South, Cranford, Wives and Daughters",
	"author": "Elizabeth Gaskell",
	"positionTop": 38,
	"positionLeft": 48,
	"townName":  "Manchester",
	"bodyText": "Gaskell's historic home was restored in 2014. It is now a museum to the author and her times."
};

describe('EnglandMap component', () => {
  it('should match snapshot', () => {
    const component = ReactTestRenderer.create(
      <EnglandMap />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('openInfoBox() should update state', () => {
    const component = Enzyme.shallow(<EnglandMap />);
    expect(component.state('infoBoxOpen')).toEqual(false);
    component.instance().openInfoBox(true, testLocation);
    expect(component.state('infoBoxOpen')).toEqual(true);
    expect(component.state('infoBoxLocation')).toEqual(testLocation);
  })
});