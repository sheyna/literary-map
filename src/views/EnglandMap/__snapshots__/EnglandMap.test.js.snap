// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnglandMap component should match snapshot 1`] = `
Array [
  <div
    className="mapElement"
  >
    <div
      className="MapElement"
    >
      <img
        alt="Map of England"
        src="map-of-england.svg"
      />
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "49%",
            "top": "30%",
          }
        }
      >
        <img
          alt="BrontÃ« Country"
          src="img/brontes-thumb.png"
        />
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "27%",
            "top": "77%",
          }
        }
      >
        <img
          alt="A Romance of Exmoor"
          src="img/lorna-doone-thumb.png"
        />
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "50%",
            "top": "58%",
          }
        }
      >
        <img
          alt="The Bard of Stratford-upon-Avon"
          src="img/shakespeare-thumb.png"
        />
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "38%",
            "top": "20%",
          }
        }
      >
        <img
          alt="William Wordsworth's home"
          src="img/wordsworth-thumb.png"
        />
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "60%",
            "top": "22%",
          }
        }
      >
        <img
          alt="Count Dracula Arrives in England"
          src="img/dracula-thumb.png"
        />
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "70%",
            "top": "72%",
          }
        }
      >
        <img
          alt="London's Writers' Block"
          src="img/cheyne-thumb.png"
        />
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "52%",
            "top": "37%",
          }
        }
      >
        <img
          alt="Romance in the English Civil War"
          src="img/peveril-peak-thumb.png"
        />
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "46%",
            "top": "40%",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-location-pin locationPin"
          data-icon="location-pin"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "48%",
            "top": "38%",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-location-pin locationPin"
          data-icon="location-pin"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "62%",
            "top": "78%",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-location-pin locationPin"
          data-icon="location-pin"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "49%",
            "top": "55%",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-location-pin locationPin"
          data-icon="location-pin"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "62%",
            "top": "83%",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-location-pin locationPin"
          data-icon="location-pin"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "45%",
            "top": "85%",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-location-pin locationPin"
          data-icon="location-pin"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "17%",
            "top": "87%",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-location-pin locationPin"
          data-icon="location-pin"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "53%",
            "top": "70%",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-location-pin locationPin"
          data-icon="location-pin"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "39%",
            "top": "85%",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-location-pin locationPin"
          data-icon="location-pin"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="MapLocationPoint"
        onClick={[Function]}
        style={
          Object {
            "left": "75%",
            "top": "80%",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-location-pin locationPin"
          data-icon="location-pin"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>,
  <div
    className="pageContents"
  >
    <p
      className="introText"
    >
      For such a small county, England has produced hundreds of the great works of literature. From Latin and Anglo-Norman literatures to the early development of the English language, stage plays and the birth of the novel, England has seen many types of great literature.
    </p>
  </div>,
]
`;
